specs/ContextMenuRequested.md

```diff
@@ -0,0 +1,764 @@
+    interface ICoreWebView2ContextMenuRequestedEventArgs : IUnknown
+    {
+        /// Gets the collection of `ContextMenuItem` objects. 
+        /// See `ICoreWebView2ContextMenuItemCollection` for more details.
+        [propget] HRESULT MenuItems([out, retval] ICoreWebView2ContextMenuItemCollection** value);
+
+        /// Gets the information associated with the requested context menu. 
+        /// See `ICoreWebView2ContextMenuInfo` for more details.
+        [propget] HRESULT ContextMenuInfo([out, retval] ICoreWebView2ContextMenuInfo** value);
+
+        /// Sets the selected command for the WebView to execute. The value is
+        /// obtained via the `ContextMenuItem` CommandId property
+        /// The default value is -1 which means that no selected occured.
+        [propput] HRESULT SelectedCommandId([in] INT32 value);
+
+        /// Gets the selected CommandId.
+        [propget] HRESULT SelectedCommandId([out, retval] INT32* value);
+
+        /// Sets whether the `ContextMenuRequested` event is handled by host after
+        /// the event handler completes or if there is a deferral then after the deferral is completed.  
+        /// If Handled is set to TRUE then WebView2 will not display a context menu and will instead
+        /// use the SelectedCommandId property to indicate which, if any, context menu item to invoke.
+        /// If after the event handler or deferral completes Handled is set to FALSE then WebView2 
+        /// will display a context menu based on the contents of the MenuItems property. 
+        /// The default value is FALSE.
+        [propput] HRESULT Handled([in] BOOL value);
+
+        /// Gets whether the `ContextMenuRequested` event is handled by host.
+        [propget] HRESULT Handled([out, retval] BOOL* value);
+
+        /// Returns an `ICoreWebView2Deferral` object. Use this operation to
+        /// complete the event when the custom context menu is closed.
+        HRESULT GetDeferral([out, retval] ICoreWebView2Deferral** deferral);
+    }
+
+    /// Represents the information regarding the context menu. Includes the location of the request, 
+    /// the context selected and the appropriate data used for the actions of a context menu.
+    [uuid(b8611d99-eed6-4f3f-902c-a198502ad472), object, pointer_default(unique)]
+    interface ICoreWebView2ContextMenuInfo : IUnknown
+    {
+        /// Gets the coordinates where the context menu request occured in relation to the upper left corner of the webview bounds.
+        [propget] HRESULT Location([out, retval] POINT* value);
+
+        /// Gets the kind of context that the user selected.
+        [propget] HRESULT ContextKind([out, retval] COREWEBVIEW2_CONTEXT_KIND* value);
+
+        /// Returns TRUE if the context menu was requested on the main frame and
+        /// FALSE if invoked on another frame.
+        [propget] HRESULT IsMainFrame([out, retval] BOOL* value);
```

Comment:
Confusingly-named, because this is a context menu info, so this presumably tells me whether the context menu is a main frame (which it never is). Maybe `IsRequestedForMainFrame`.

It seems that a lot of these properties are really properties on the context menu context, not properties on the context menu. So maybe we should move the context menu properties to the args, and call this ContextMenuContextInfo? Then the properties more clearly are describing the context menu context, not the context menu.

Also, maybe we can come up with a different word from "context" to avoid the confusion between "context menu' and "context menu context".

Reply:
I agree with moving the context menu properties onto the event args and keeping context menu context properties in their own class named with Source or Target or other similar sort of names (does the browser extensions API have any similar names?)

Reply:
@david-risney the browser extensions API calls it "OnClickData" and it provides most of the information we will provide in our ContextMenuTarget object.